cmd/ttn-lw-cli/commands/end_devices.go

```diff
@@ -349,13 +349,19 @@ var (
 			}
 			paths := util.UpdateFieldMask(cmd.Flags(), setEndDeviceFlags, attributesFlags())
 
+			abp, _ := cmd.Flags().GetBool("abp")
+			multicast, _ := cmd.Flags().GetBool("multicast")
 			var device ttnpb.EndDevice
 			if inputDecoder != nil {
 				decodedPaths, err := inputDecoder.Decode(&device)
 				if err != nil {
 					return err
 				}
 				paths = append(paths, ttnpb.FlattenPaths(decodedPaths, endDeviceFlattenPaths)...)
+
+				if ttnpb.ContainsField("supports_join", paths) {
+					abp = !device.SupportsJoin
```

Comment:
We should also error if `abp` or `multicast` is set, but `device.SupportsJoin == true`

Reply:
We can do that, but I think is that the device fields from JSON input should just take precedence over the `apb` and `multicast` flags when creating multiple devices, right?

Reply:
In general, if the input explicitly specifies `"supports_join": true` then we should be creating an OTAA device.

Reply:
Yes, but if the user specified `abp` or `multicast` by flag, then that is an error

Reply:
Change to raise an `ambiguous_supports_join` error in that case.